presto-main/src/main/java/io/prestosql/server/ui/FormWebUiAuthenticationManager.java

```diff
@@ -338,7 +392,7 @@ static String getRedirectLocation(HttpServletRequest request, String path, Strin
 
     private boolean isAuthenticationEnabled(HttpServletRequest request)
     {
-        return !isSecure(request, httpsForwardingEnabled) || passwordAuthenticatorManager.isLoaded();
+        return !isSecure(request, httpsForwardingEnabled) || passwordAuthenticatorManager.isLoaded() || authenticator.isPresent();
```

Comment:
Why is the first one `!isSecure()`?

Reply:
unsecured requests support username-only authentication (no password).  I added some comments